Generate-Random-number/script.js

```diff
@@ -16,290 +351,281 @@ const mongoDB = process.env.MONGODB_URI;
+    const phoneRegex = new RegExp(`.*${phone}.*`, "i");
+
+    const skip = (page - 1) * numericLimit;
 
-        // Perform the search query with pagination
-        const users = await User.find(searchQuery)
-            .skip(skip)  // Skip the first 'skip' documents
-            .limit(numericLimit)  // Limit the result to 'limit' documents
-            .exec();  // Execute the query
-
-            console.log('Search query:', searchQuery);
-        console.log('Users found:', users);
-
-        // Count the total number of documents matching the search query
-        const count = await User.countDocuments(searchQuery);
-         console.log('count' , count)
-        // Calculate total pages
-        /*Total Users (count):100
-             Users Per Page (numericLimit): 20    100 / 20 = 5*/
-        const totalPages = Math.ceil(count / numericLimit);
-        console.log('totalPages', totalPages)
-        // Respond with the list of users, total pages, and current page
-        res.json({ users, totalPages, currentPage: Number(page) });
-    } catch (error) {      
-        // Handle errors
-        console.error(error);
-        res.status(500).json({ error: 'Server error' });
+    const searchQuery = {};
+
+    if (name) {
+      searchQuery.username = { $regex: nameRegex };
     }
-};
- const deleteUsers = async (req, res) => {
-    const batchSize = 100; // Number of records to delete per batch
-
-    try {
-        let totalDeleted = 0;
-        let deletedInCurrentBatch = 0;
-
-        do {
-            // Fetch a batch of users to delete
-            const usersToDelete = await User.find({}).limit(batchSize);
-
-            // If no more users to delete, break out of the loop
-            if (usersToDelete.length === 0) {
-                break;
-            }
-            // Delete the batch of users
-            const result = await User.deleteMany({ _id: { $in: usersToDelete.map(user => user._id) } });
-            deletedInCurrentBatch = result.deletedCount;
-            totalDeleted += deletedInCurrentBatch;
-
-            console.log(`Deleted ${totalDeleted} records so far`);
-
-            // Optional: Add a delay to avoid overwhelming the server
-            if (deletedInCurrentBatch > 0) {
-                await new Promise(resolve => setTimeout(resolve, 100));
-            }
-        } while (deletedInCurrentBatch === batchSize); // Continue while the batch size is met
-
-        res.status(200).json({ message: `Deleted ${totalDeleted} users` });
-    } catch (error) {
-        console.error('Error deleting users:', error);
-        res.status(500).send('Error deleting users');
+
+    if (email) {
+      searchQuery.email = { $regex: emailRegex };
+    }
+
+    if (phone) {
+      searchQuery.phone = { $regex: phoneRegex };
     }
+
+    const users = await User.find(searchQuery)
+      .skip(skip)
+      .limit(numericLimit)
+      .exec();
+
+    console.log("Search query:", searchQuery);
```

Comment:
why is this search user function here? this should be in the real react and express app where you and laiba created forms